src/probe_matcher.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include "ast.h"
+
+#include <set>
+#include <sstream>
+
+#include <linux/perf_event.h>
+
+namespace bpftrace {
+
+const std::string kprobe_path =
+    "/sys/kernel/debug/tracing/available_filter_functions";
+const std::string tp_avail_path = "/sys/kernel/debug/tracing/available_events";
+const std::string tp_path = "/sys/kernel/debug/tracing/events";
+
+struct ProbeListItem
+{
+  std::string path;
+  std::string alias;
+  uint32_t type;
+  uint64_t defaultp;
+};
+
+// clang-format off
+const std::vector<ProbeListItem> SW_PROBE_LIST = {
```

Comment:
I guess we need to use a singleton to thread-safe initialization. something like: https://github.com/iovisor/bpftrace/blob/77a25cf9eb76f1107240ee26cca8b7223bb5184f/src/ast/semantic_analyser.cpp#L22-L34
@danobi probably know this.